llama_index/llms/xinference.py

```diff
@@ -0,0 +1,153 @@
+from typing import Any, Dict, Sequence
+
+from llama_index.constants import DEFAULT_NUM_OUTPUTS
+from llama_index.llms.base import (
+    ChatMessage,
+    ChatResponse,
+    ChatResponseGen,
+    CompletionResponse,
+    CompletionResponseGen,
+    LLMMetadata,
+    MessageRole,
+)
+from llama_index.llms.custom import CustomLLM
+from llama_index.llms.xinference_utils import (
+    xinference_message_to_history,
+    xinference_modelname_to_contextsize,
+)
+
+# an approximation of the ratio between llama and GPT2 tokens
+TOKEN_RATIO = 2.5
```

Comment:
I'm guessing this is because all our token counting is done with gpt2 tiktoken?

2.5 seems like a huge number though, tbh I didn't realize it would be that different :she

Reply:
Or is this just here because Xorbits doesn't provide a way to set the max content window? 🤔 So it's accounting for the default of 4096?

Reply:
Yes, the gpt2 tiktoken gave birth to this constant, and 2.5 is much larger than we expected as well. It is being set as an upper bound to make sure there are no failures for all possible texts and models, so the users never have to deal with mysterious failures that appear and disappear seemingly at random. The largest ratio we've seen is around ~2.3.

Ideally, we can refactor the code so that the tokenizer changes based on the model used, but for a first pull request we thought we might just use a temporary workaround instead of messing with the global structure.

To answer your second question, the user can control the max-content window when initiating it with Xorbits, so that is not the main issue. The window size can be set using the `n_ctx` parameter. We have added an example setting the window size as 4096 in the demo notebook.